### ISSUE TITLE
comma-spacing has false positive when paranthesis are used

### ISSUE THREAD

**Issue by trichins** [eslint/eslint#1457]
If you turn on comma-spacing and then lint the following file:

``` js
(function () {
    'use strict';
    return [ (''), '' ];
}());
```

You'll get a message saying there should be no space before ','.  Looking at the list of nodes that the rule is seeing, it appears that the parenthesis are stripped out of the nodes and so it just sees the two quoted strings.

It should not ignore the parenthesis and it should treat this as valid.
